blocksync/blocksync.go

```diff
@@ -293,14 +291,17 @@ func (bs *blockSyncer) ProcessBlock(ctx context.Context, peer string, blk *block
 	log.L().Debug("flush blocks", zap.Uint64("start", tip), zap.Uint64("end", syncedHeight))
 	if syncedHeight > bs.lastTip {
 		bs.lastTip = syncedHeight
-		bs.lastTipUpdateTime = time.Now()
 	}
-	requestMaxHeight := atomic.LoadUint64(&bs.requestMaxHeight)
+	bs.checkSync(syncedHeight)
+	return nil
+}
+
+func (bs *blockSyncer) checkSync(syncedHeight uint64) {
+	requestMaxHeight := atomic.LoadUint64(&bs.lastRequestHeight)
```

Comment:
`lastReqHeight := `

Reply:
also, merge above 3 lines of code into here
```
if syncedHeight > bs.lastTip {
		bs.lastTip = syncedHeight
		bs.lastTipUpdateTime = time.Now()
}
```
lastTip (and updateTime) is related to decide if we should send sync request

Reply:
also an earlier comment
> think we should also use atomic.SwapUint64 to handle bs.lastTip